### PR TITLE
Use the window icon first when choosing border/taskbar icons.

### LINKED ISSUE #30: Window icons should use the window icon before falling back to the application icon

### PR DESCRIPTION
**Pull request by okratitan**
Fixes #30

### CODE REVIEW

**Comment by andydotxyz** [comment]
Works well and code is clean. Happy to accept, however as discussed on Slack should we perhaps fall back to theme or app in the taskbar so it's not all pixelated?

**Comment by andydotxyz** [comment]
oh man I think there is something awry with the nil checking - we now get missing icons instead of the missing icon icon...

File: wm/client.go [hunk 1]
```diff
@@ -65,6 +65,15 @@ func (c *client) IconName() string {
 	return windowIconName(c.wm.x, c.win)
 }
 
+func (c *client) Icon() fyne.Resource {
+	var icon fyne.Resource = nil
```
Comment (andydotxyz) [inline wm/client.go]: This might be your problem... try moving the return into the if statement - the icon var is not needed...

**Comment by okratitan** [comment]
Of note - I think transparency may be broken... It's weird.

**Comment by andydotxyz** [comment]
Yup, good to merge as soon as Travis is happy (gofmt error)

**Pull request merged**